system/jlib/jlzw.cpp

```diff
@@ -2032,14 +2034,40 @@ class CCompressedFile : implements ICompressedFileIO, public CInterface
     void getblock(offset_t pos)
     {
         curblockbuf.clear();
-        if (pos == nextExpansionPos)
+
+        //If the blocks are being expanded incrementally check if the position is within the current block
+        //This test will never be true for row compressed data, or non-incremental decompression
+        if ((pos >= startBlockPos) && (pos < startBlockPos + fullBlockSize))
         {
-            size32_t nextSize = expander->expandNext(curblockbuf);
-            if (nextSize != 0)
+            if (pos < nextExpansionPos)
             {
-                curblockpos = pos;
-                nextExpansionPos = pos+nextSize;
-                return;
+                //Start decompressing again and avoid re-reading the data from disk
+                void * rawData;
+                if (fileio)
+                    rawData = compressedInputBlock.mem();
```

Comment:
could/should use a const void * here , and compressedInputBlock.get() ?

Reply:
Yes, I will change